### DISCUSSION TITLE
formatStateUsing() ignores copyable().

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#12646] [Help] [answered]
### Package

Table builder

### Package Version

v3.2.73

### How can we help you?

I want to copy the original text when I click the TextColumn . 

If formatStateUsing() is used together with copyable(), the copyable() does not work.

```php
Tables\Columns\TextColumn::make('address')
    ->copyable()
    ->formatStateUsing(function ($state) {
        if (strlen($state) > 10) {
            return substr($state, 0, 6) . '...' . substr($state, -4);
        }
        return $state;
    }),


Tables\Columns\TextColumn::make('address')
    ->formatStateUsing(function ($state) {
        if (strlen($state) > 10) {
            return substr($state, 0, 6) . '...' . substr($state, -4);
        }
        return $state;
    })    
    ->copyable(),

```

Both order of the functions does not work.

Is there a way to use both formatStateUsing() and copyable() together?

**Comment by magic-thomas** [comment, marked as answer]
I found the reason.. 

copyable() does not work on http. It works on https only. 

seems livewire limit.

Because I use Laravel Herd and didn't set it https, it happened. 

HTTPS only for copyable(). 

Hope it helps to someone.

**Reply by ramiropiro** [reply]
thanks!
